api/lib/infrastructure/utils/query-params-utils.js

```diff
@@ -35,3 +37,7 @@ function _extractObjectParameter(query, regex) {
 function _extractArrayParameter(query, parameterName) {
   return query[parameterName] ? query[parameterName].split(',') : [];
 }
+
+function _convertToInt(params) {
+  return _.mapValues(params, (value) => parseInt(value));
```

Comment:
Quand le paramètre d'une fonction et le paramètre de la suivante on peut l'annuler 😄 A moins que les paramètres suivants influence le `parseInt` 🤔 ?

```suggestion
  return _.mapValues(params, parseInt);
```

Reply:
Surtout pas, parce que `parseInt` prend en deuxième argument la base :

```
> ['10','10','10'].map(parseInt)
[ 10, NaN, 2 ]
```

Avec `_.mapValues` c'est un peu moins probable de faire un bug (il faut que l'objet ait une clé numérique,  ex: `{ '2': '123 }` mais c'est quand même pas recommandé…

Sinon il y a `_.toInteger` qui n'a pas ce problème 🤔 

Au passage je ne trouve pas le nom de cette fonction très parlant : `_convertToInt` alors qu'en fait ça convertit en entier _les valeurs d'un objet_. On pourrait lui trouver un meilleur nom mais vu qu'elle s'écrit `_.mapValues(…, _.toInteger)` je serais d'avis de garder simplement ça dans `extractPage`.

Reply:
+1

Reply:
J'ai remplacé le parseInt par le toInteger. Je trouve que l'extraire dans une méthode donne un meilleur niveau d'abstraction même si effectivement c'est pas un gros gain par rapport  à `_.mapValues(…, _.toInteger)`